easybuild/tools/build_from_specsfile.py

```diff
@@ -0,0 +1,150 @@
+    def parse(filename):
+        raise NotImplementedError
+
+
+class YamlSpecParser(GenericSpecsParser):
+    @staticmethod
+    def parse(filename):
+
+        try:
+            with open(filename, 'r') as f:
+                spec_dict = yaml.safe_load(f)
+
+            eb = Specsfile()
+        except FileNotFoundError:
+            raise EasyBuildError("Could not read provided specsfile.")
+
+        sw_dict = spec_dict["software"]
+
+        # assign software-specific EB attributes
+        for software in sw_dict:
+            try:
+                # iterates through toolchains to find out what sw version is needed
+                for yaml_toolchain in sw_dict[software]['toolchains']:
+                    # retrieves version number
+                    for yaml_version in sw_dict[software]['toolchains'][yaml_toolchain]['versions']:
+
+                        # if among versions there is anything else than known flags or version flag itself, it is wrong
+                        # identification of version strings is vague
+                        if str(yaml_version)[0].isdigit() \
+                                or str(yaml_version)[-1].isdigit():
+                            # creates a sw class instance
+                            try:
+                                yaml_toolchain_name = str(yaml_toolchain).split('-', 1)[0]
+                                yaml_toolchain_version = str(yaml_toolchain).split('-', 1)[1]
+                            except IndexError:
+                                yaml_toolchain_name = str(yaml_toolchain)
+                                yaml_toolchain_version = ''
+
+                            sw = SoftwareSpecs(software=software, version=yaml_version,
+                                               toolchain=yaml_toolchain, toolchain_name=yaml_toolchain_name,
+                                               toolchain_version=yaml_toolchain_version)
+
+                            # append newly created class instance to the list inside EbFromSpecs class
+                            eb.software_list.append(sw)
+
+                        elif str(yaml_version) == 'versionsuffix':
+                            try:
+                                version_info = sw_dict[software]['toolchains'][yaml_toolchain]['versions'][yaml_version]
+                                if version_info['versionsuffix'] is not None:
+                                    sw.version_suffix = version_info['versionsuffix']
+                            except (KeyError, TypeError, IndexError):
+                                continue
+                        elif str(yaml_version) == 'exclude-labels' \
+                                or str(yaml_version) == 'include-labels':
+                            continue
+
+                        else:
+                            print('Software % s has wrong yaml structure!' % (str(software)))
+                            raise EasyBuildError('Wrong yaml structure')
```

Comment:
Provide some more info in the error message: what was being parsed when this happened? Which software name for example...